WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/Time Range/AnalyticsHubDayRangeSelection.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+final class AnalyticsHubDayRangeSelection: AnalyticsHubTimeRangeSelectionDelegate {
+    var currentTimeRange: AnalyticsHubTimeRange?
+    var previousTimeRange: AnalyticsHubTimeRange?
+    var currentRangeDescription: String?
+    var previousRangeDescription: String?
+
+    init(referenceDate: Date, currentCalendar: Calendar) {
+        let currentTimezone = currentCalendar.timeZone
+
+        let currentDayStart = referenceDate.startOfDay(timezone: currentTimezone)
+        let currentTimeRange = AnalyticsHubTimeRange(start: currentDayStart, end: referenceDate)
+        self.currentTimeRange = currentTimeRange
+        self.currentRangeDescription = DateFormatter.Stats.analyticsHubDayMonthYearFormatter.string(from: currentTimeRange.start)
+
+        if let previousDay = currentCalendar.date(byAdding: .day, value: -1, to: referenceDate) {
+            let previousDayStart = previousDay.startOfDay(timezone: currentTimezone)
+            let previousTimeRange = AnalyticsHubTimeRange(start: previousDayStart, end: previousDay)
+            self.previousTimeRange = previousTimeRange
+            self.previousRangeDescription = DateFormatter.Stats.analyticsHubDayMonthYearFormatter.string(from: previousTimeRange.start)
+        } else {
+            self.previousTimeRange = nil
+            self.previousRangeDescription = nil
+        }
```

Comment:
Q: is there a case to have correct `currentTimeRange`, but `previousTimeRange = nil`?

My idea is: if partial fail means inability to display relevant information to the user, we can make RangeSelection initializers failable. So if something goes wrong in init phase - fail and throw readable error instead of creating a data object with all values = nil.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8267/commits/1077d07c3186a793daffece808c0132f1cc787cb